quinn-proto/src/token.rs

```diff
@@ -185,6 +202,45 @@ pub(crate) struct IncomingToken {
     pub(crate) orig_dst_cid: ConnectionId,
 }
 
+impl IncomingToken {
+    /// Construct for an `Incoming` which is not validated by a token
+    fn unvalidated(header: &InitialHeader) -> Self {
+        Self {
+            retry_src_cid: None,
+            orig_dst_cid: header.dst_cid,
+        }
+    }
+
+    /// Construct for an `Incoming` given the first packet header, or error if the connection
+    /// cannot be established
+    pub(crate) fn from_header(
+        header: &InitialHeader,
+        server_config: &ServerConfig,
+        remote_address: SocketAddr,
```

Comment:
Note: Most logic in this module is passing around `SocketAddr` by reference. However, in my opinion, for small `Copy` types, one should just pass them by value and let the compiler figure out what it wants to do.

Reply:
Yup, we should probably rely on `Copy` more for this.

Reply:
Great. I will likely create a "refactor token encoding" PR after this one that will do that among other things.